src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/GEOSoutputincr_GridComp/GEOS_OutputIncrGridComp.F90

```diff
@@ -0,0 +1,391 @@
+#include "MAPL_Generic.h"
+
+!=============================================================================
+module GEOS_OutputIncrGridCompMod
+
+  !BOP
+  ! !DESCRIPTION:
+  !
+  !   {\tt Obs} is a gridded component to
+  !   {\tt Obs} has no children.
+  
+  !
+  ! !USES:
+  
+  use ESMF
+  use MAPL_Mod
+  use ESMF_CFIOMOD,              only: ESMF_CFIOstrTemplate
+  
+  implicit none
+
+  include 'mpif.h'
+  
+  private
+  ! !PUBLIC MEMBER FUNCTIONS:
+  public :: SetServices
+  !
+  !EOP
+  !
+  integer, parameter :: NUM_SUBTILES = 4
+
+contains
+  
+  ! ******************************************************************************
+
+  !BOP
+  ! !IROUTINE: SetServices -- Sets ESMF services for component
+  ! !INTERFACE:
+  
+  subroutine SetServices ( GC, RC )
+    
+    ! !ARGUMENTS:
+    
+    type(ESMF_GridComp),intent(INOUT) :: GC
+    integer, optional,  intent(  OUT) :: RC
+    
+    ! !DESCRIPTION:
+    
+    !EOP
+    !
+    ! ErrLog Variables
+    
+    character(len=ESMF_MAXSTR)   :: Iam
+    character(len=ESMF_MAXSTR)   :: COMP_NAME
+    integer                      :: STATUS
+    
+    ! Local Variables
+    type(MAPL_MetaComp), pointer :: MAPL=>null()
+    type(ESMF_Config)            :: CF
+
+    ! Begin...
+    ! --------
+    
+    ! Get my name and set-up traceback handle
+    ! ------------------------------------------------------------------------------
+    
+    Iam='SetServices'
+    call ESMF_GridCompGet ( GC, NAME=COMP_NAME, RC=STATUS )
+    _VERIFY(STATUS)
+    Iam=trim(COMP_NAME)//trim(Iam)
+    
+    ! Register services for this component
+    call MAPL_GridCompSetEntryPoint(                                            &
+         gc,                                                                    &
+         ESMF_METHOD_INITIALIZE,                                                &
+         Initialize,                                                            &
+         rc=status                                                              &
+         )
+    _VERIFY(status)
+    
+    call MAPL_GridCompSetEntryPoint(                                            &
+         gc,                                                                    &
+         ESMF_METHOD_FINALIZE,                                                  &
+         Finalize,                                                              &
+         rc=status                                                              &
+         )
+    _VERIFY(status)
+        
+  ! Exports for Catchment prognostics increments
+
+  call MAPL_AddExportSpec(GC                                                      ,&
```

Comment:
This list of calls to MAPL_AddExportSpec() for the increments is also present verbatim in GEOS_LandAssimGridComp.F90 (beginning ~line 227).
I'm guessing this child GC F90 file is the right place to define the list (w/ LONG_NAME and UNITS).
But do we need to repeat this list in GEOS_LandAssimGridComp.F90?
Maybe we can use the syntax that is used in GEOS_LandGridComp.F90 to bubble up the exports from Catch to Land?  That is, can we replace the entries here with something like
`   call MAPL_AddExportSpec(GC, SHORT_NAME = 'TCFSAT_INCR', CHILD_ID = CATCH(1), RC=STATUS); VERIFY_(STATUS)
`
etc?
This might not work because here we're dealing with ensemble members, but maybe it does.  If so, it would be good to have to define LONGNAME and UNITS in only one place.

Reply:
I don't think we can do that. The export variables in LandAssim  are different from those in  ExportCatchIncr even though the names are the same (Sara told me the name should be the same). If there are no collections in HISTORY.rc to export the incr of individual member , the export won't be allocated and initialized.